### ISSUE TITLE
Euphonic Calculations: Performance Benchmarking: Initial Implementation

### ISSUE THREAD

**Issue by JamesKingWork** [pace-neutrons/Pace-Project-Plan#122]
Sub issue of #26 

As a user of Euphonic, I would like the package to be performance benchmarked to ensure its performance doesn't degrade over time

Notes:

- Becky is providing .castep_bin files for the materials, a file with randomly generated q points and a scarf .sbatch file used to run jobs
- There may be issues with `pip install .` on SCARF
- See https://github.com/pace-neutrons/pace-developers/blob/master/euphonic/testing/01_performance_testing_discussion.md

**Comment by JamesKingWork** [comment]
Unfortunately, due to Anvil's SCARF node being down and not being able to contact them the running of benchmark tests on Anvil is currently not possible. As a result this ticket is being reduced to creating the scripts to generate test data and the test scripts themselves (with a Jenkinsfile that will hopefully work). I will spin out another ticket to do the running on Anvil itself which can be picked up when ready.

**Comment by JamesKingWork** [comment]
https://github.com/pace-neutrons/Euphonic/pull/59

**Comment by JamesKingWork** [comment]
Acceptance criteria:

- [ ] A Jenkins pipeline is set up to performance test `InterpolationData.calculate_fine_phonons` on SCARF
- [ ] Testing runs with 1 core on Python and 1, 2, 4, 12, 24 cores on C
- [ ] Testing is run against a large number of randomly generated q-points (10,000 initially)
- [ ] Testing is run against a few different materials
- [ ] Only master is run and only once a week to prevent SCARF overload

**Comment by JamesKingWork** [comment]
Jenkins job running the benchmarking is here: https://anvil.softeng-support.ac.uk/jenkins/job/PACE-neutrons/job/Euphonic/job/Performance%20Benchmarking/
